homeassistant/components/alert/__init__.py

```diff
@@ -196,6 +177,24 @@ def state(self) -> str:
             return STATE_ON
         return STATE_IDLE
 
+    async def async_added_to_hass(self) -> None:
+        """Add hass to templates and register for tracking state changes."""
+        if self._message_template is not None:
+            self._message_template.hass = self.hass
+        if self._done_message_template is not None:
+            self._done_message_template.hass = self.hass
+        if self._title_template is not None:
+            self._title_template.hass = self.hass
+
+        self._unsub = async_track_state_change_event(
```

Comment:
You can wrap this on self.async_on_remove instead

Reply:
Thanks for that example, that's helpful. 

I think that would be good suggestion if we only unsubscribe when the alert is removed, but note that we also unsubscribe/resubscribe in `async_update_config`, in case the entity that is being watched changes. So I believe I need to maintain the unsubscribe pointer so I can call it in that case?

Reply:
You could do like this, then you don't need to remember unsubscribing from `async_will_remove_from_hass`
```py
        self._unsub = async_track_state_change_event(
            self.hass, [self._watched_entity_id], self.watched_entity_change
        )
        self.async_on_remove(self._unsub)
```

Reply:
I feel like this will not work?

If I push self._unsub to the _on_remove list, and then later call the reload function with updated yaml (say changing the watched entity), I then need to unsubscribe and register for a new watched_entity_change, and save the new unsubscribe function. 

But the original unsub function will still be in the _on_remove list. I would need some way to remove that function from the _on_remove list and replace it with the new one when I update the entity. I don't see a good way to do this.

Given that, I think I prefer it how it is?